### PR TITLE
[Frameworks] Support LGBM auto-logging

### PR DESCRIPTION
**Pull request by AlxZed**

### CODE REVIEW

File: mlrun/frameworks/lightgbm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# Temporary placeholder, LGBMModelServer may deviate from PklModelServer in upcoming versions.
+LGBMModelServer = PickleModelServer
+
+
+def apply_mlrun(
+    model,
+    context: mlrun.MLClientCtx = None,
+    X_test=None,
+    y_test=None,
+    model_name=None,
```
Comment (guy1992l) [inline mlrun/frameworks/lightgbm/__init__.py:19]: Take a look at the tf_keras or pytorch code, there are typing hints for every known type. For example: model_name: str.

Remember, the better the code is written and documented, the less questions you will receive.

File: mlrun/frameworks/lightgbm/__init__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    generate_test_set=True,
+    feature_vector = None,
+    **kwargs
+):
+    """
```
Comment (guy1992l) [inline mlrun/frameworks/lightgbm/__init__.py:24]: Make sure the parameters are aligned to the order of the parameters in the function's header. Also, please align the spaces.

**Review by guy1992l — CHANGES_REQUESTED** [review]
Good job Alex, straight forward implementation. But, it's the 3rd framework using the MLBaseMLRunInterface which is not as I meant it to be. Before the code is getting out of hand, you should implement ModelHandler, some type of logger and work with them as well. The MLRunInterface should only wrap the model's functions, and not log or save any models and / or artifacts. Each class has its own purpose and job, this way you keep the code organized, readable, generic and modular.

**Comment by guy1992l** [comment]
Also, rename xgboost to xgb and sklearn to scikit_learn

**Comment by AlxZed** [comment]
> Also, and very important, change the directory name from lightgbm to lgbm. It is very unrecommended to use the same module name as the original framework, imagine the user will do:
> 
> import lightgbm from mlrun.frameworks import lightgbm
> 
> This is not a good scenario

People use lgbm as the following:
`import lightgbm as lgb`
or, more rarely:
`import lightgbm as lgbm`

if we replace it to lgbm, a user might do:
`import lightgbm as lgbm`
`from mlrun.frameworks import lgbm`

File: mlrun/frameworks/lgbm/__init__.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+"""
```
Comment (Hedingber) [inline mlrun/frameworks/lgbm/__init__.py:1]: You added a new module - `mlrun/frameworks/lgbm` - need to add it in `setup.py`

File: tests/test_mlbase.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import lightgbm as lgb
```
Comment (Hedingber) [inline tests/test_mlbase.py:1]: Please add the tests under `tests/frameworks` - either in the already existing `test_mlbase.py` file or in a new `test_lgbm.py` file

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: dockerfiles/jupyter/requirements.txt [hunk 5]
```diff
@@ -4,6 +4,7 @@ scikit-learn~=0.23.0
 seaborn~=0.11.0
 scikit-plot~=0.3.7
 xgboost~=1.1
+lightgbm~=2.3.0
```
Comment (yaronha) [inline dockerfiles/jupyter/requirements.txt:7]: why is it added specifically to the jupyter image?

File: tests/frameworks/test_mlbase.py [hunk 6]
```diff
@@ -22,36 +30,76 @@ def get_dataset(classification=True):
...
-    assert model.metrics["accuracy"] > 0, "wrong accuracy"
-    assert model.model_file == "LogisticRegression.pkl"
+    assert (sklearn_run.artifact("model").meta.to_dict()["metrics"]["accuracy"]) > 0
+    assert (
+        sklearn_run.artifact("model").meta.to_dict()["model_file"]
+    ) == "LogisticRegression.pkl"
 
 
-@pytest.mark.skipif(skip_tests, reason="missing packages")
+@pytest.mark.skipif(_is_installed("xgboost"), reason="missing packages")
```
Comment (Hedingber) [inline tests/frameworks/test_mlbase.py:87]: be more specific, same for other tests
```suggestion
@pytest.mark.skipif(_is_installed("xgboost"), reason="xgboost package missing")
```

File: tests/frameworks/test_mlbase.py [hunk 7]
```diff
@@ -22,36 +30,76 @@ def get_dataset(classification=True):
...
+        pass
 
-    model = xgb.XGBRegressor()
-    X_train, X_test, y_train, y_test = get_dataset(classification=False)
-    model = apply_mlrun_xgb(model, context, X_test=X_test, y_test=y_test)
-    model.fit(X_train, y_train)
+
+def run_mlbase_lgbm_classification(context):
+    try:
+        import lightgbm as lgb
```
Comment (Hedingber) [inline tests/frameworks/test_mlbase.py:62]: The import is happening only when the function is called, the function will be called only when the test will run, the test should run only when the package exists, so there's no reason to wrap with try except here (and in `run_mlbase_xgboost_regression`)

File: mlrun/frameworks/_common/plots.py [hunk 8]
```diff
@@ -4,9 +4,22 @@
...
+
+try:
+    from scikitplot.metrics import plot_calibration_curve
+except ModuleNotFoundError:
+    pass
+
+try:
+    from scipy import interp
+except ModuleNotFoundError:
+    pass
```
Comment (Hedingber) [inline mlrun/frameworks/_common/plots.py:21]: Why it's needed to wrap these with try except ?

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/frameworks/_common/plots.py (hunk 8 above)
Comment (AlxZed) [reply mlrun/frameworks/_common/plots.py:21]: They are failing due to tests, this whole function will be re-written in the next PR

File: tests/frameworks/test_mlbase.py (hunk 7 above)
Comment (AlxZed) [reply tests/frameworks/test_mlbase.py:62]: this is importing lightgbm, the xgboost function imports xgb, two different libraries. 
Importing them within the function is the only way I was able to make the dockerized tests pass

**Review by Hedingber — APPROVED** [review]

**Pull request merged**